core/device_hooks/common_kernels.inc.cpp

```diff
@@ -77,26 +77,37 @@
 
 #define GKO_STUB(_macro) _macro GKO_NOT_COMPILED(GKO_HOOK_MODULE)
 
-#define GKO_STUB_VALUE_CONVERSION(_macro)                             \
-    template <typename SourceType, typename TargetType>               \
-    _macro(SourceType, TargetType) GKO_NOT_COMPILED(GKO_HOOK_MODULE); \
-    GKO_INSTANTIATE_FOR_EACH_VALUE_CONVERSION(_macro)
 
 #define GKO_STUB_NON_COMPLEX_VALUE_TYPE(_macro)          \
     template <typename ValueType>                        \
     _macro(ValueType) GKO_NOT_COMPILED(GKO_HOOK_MODULE); \
     GKO_INSTANTIATE_FOR_EACH_NON_COMPLEX_VALUE_TYPE(_macro)
 
+#define GKO_STUB_NON_COMPLEX_VALUE_TYPE_WITH_HALF(_macro) \
+    template <typename ValueType>                         \
+    _macro(ValueType) GKO_NOT_COMPILED(GKO_HOOK_MODULE);  \
+    GKO_INSTANTIATE_FOR_EACH_NON_COMPLEX_VALUE_TYPE_WITH_HALF(_macro)
```

Comment:
I should have asked it in the previous PR, but wouldn't it be better if the instantiation with half would be the standard name and excluding it has its special suffix (`_WITHOUT_HALF`)? That would make it clear that the default is with half precision.